src/analysis.zig

```diff
@@ -1610,8 +1610,15 @@ fn resolveTypeOfNodeUncached(analyser: *Analyser, node_handle: NodeWithHandle) e
                 .unwrap_optional => try analyser.resolveOptionalUnwrap(base_type),
                 .array_access => try analyser.resolveBracketAccessType(base_type, .Single),
                 .@"orelse" => {
-                    const type_right = try analyser.resolveTypeOfNodeInternal(.{ .node = datas[node].rhs, .handle = handle }) orelse return null;
-                    return try analyser.resolveOrelseType(base_type, type_right);
+                    const type_right = try analyser.resolveTypeOfNodeInternal(.{ .node = datas[node].rhs, .handle = handle });
+                    if (type_right) |tr| {
+                        return try analyser.resolveOrelseType(base_type, tr);
+                    }
+                    const type_left = try analyser.resolveTypeOfNodeInternal(.{ .node = datas[node].lhs, .handle = handle }) orelse return null;
+                    return try analyser.resolveOrelseType(base_type, switch (type_left.data) {
+                        .optional => |o| o.*,
+                        else => type_left,
+                    });
```

Comment:
The left hand side of the expression is actually being resolved in the old code. See the definition of `base_type`. `type_left` and `base_type` are the same which means that the code should be equivalent to:
```suggestion
                    const type_right = try analyser.resolveTypeOfNodeInternal(.{ .node = datas[node].rhs, .handle = handle }) orelse return try analyser.resolveOptionalUnwrap(base_type);
                    return try analyser.resolveOrelseType(base_type, type_right);
```

Reply:
Thank you for the suggestion!

Applied.